src/main/scala/io/iohk/ethereum/blockchain/sync/regular/ExecutionSync.scala

```diff
@@ -0,0 +1,142 @@
+package io.iohk.ethereum.blockchain.sync.regular
+
+import akka.actor.{Actor, ActorLogging, ActorRef, AllForOneStrategy, Props, Scheduler, SupervisorStrategy}
+import io.iohk.ethereum.blockchain.sync.{Blacklist, SyncProtocol}
+import io.iohk.ethereum.blockchain.sync.regular.RegularSync.{NewCheckpoint, ProgressState}
+import io.iohk.ethereum.consensus.validators.BlockValidator
+import io.iohk.ethereum.db.storage.StateStorage
+import io.iohk.ethereum.domain.{Block, Blockchain, BlockchainReader}
+import io.iohk.ethereum.ledger.{BlockImport, BranchResolution}
+import io.iohk.ethereum.network.Peer
+import io.iohk.ethereum.network.PeerEventBusActor.SubscriptionClassifier.MessageClassifier
+import io.iohk.ethereum.network.p2p.messages.Codes
+import io.iohk.ethereum.utils.ByteStringUtils
+import io.iohk.ethereum.utils.Config.SyncConfig
+import io.iohk.ethereum.network.PeerEventBusActor.{PeerSelector, Subscribe}
+
+class ExecutionSync(
+    fetcherService: FetcherService,
+    blockImporterService: BlockImporterService,
+    peersClient: ActorRef,
+    etcPeerManager: ActorRef,
+    peerEventBus: ActorRef,
+    blockImport: BlockImport,
+    blockchain: Blockchain,
+    blockchainReader: BlockchainReader,
+    stateStorage: StateStorage,
+    branchResolution: BranchResolution,
+    blockValidator: BlockValidator,
+    blacklist: Blacklist,
+    syncConfig: SyncConfig,
+    ommersPool: ActorRef,
+    pendingTransactionsManager: ActorRef,
+    scheduler: Scheduler
+) extends Actor
+    with ActorLogging {
+
+  val broadcaster: ActorRef = context.actorOf(
+    BlockBroadcasterActor
+      .props(new BlockBroadcast(etcPeerManager), peerEventBus, etcPeerManager, blacklist, syncConfig, scheduler),
+    "block-broadcaster"
+  )
+
+  override def receive: Receive = running(
+    ProgressState(initialBlock = 0, currentBlock = 0, bestKnownNetworkBlock = 0)
+  )
+
+  def running(progressState: ProgressState): Receive = {
```

Comment:
So this class is aimed to be a drop-in replacement for `RegularSync`?

I have some questions / remarks:
1. do you have some ideas on how we will work towards being able to replace the `RegularSync` actor in the future. ie. what needs to be done before we "flip the switch"?
2. related: so you listed the incoming messages (with similar "context.become") but what about the outgoing messages that are required?
3. can we try to keep the actor-code minimal from the get go? Ie. put the actual implementation in separate object methods (or classes if we have to.) And the actor state could then be a case class / ADT with transition methods, unit-testable without involving an actor system.
4. related to that: do you think it'd be feasible to move code out of the `RegularSync` actor and replace them piece by piece? Since in the end the API of the actor should be largely the same as we refactor it. (Assuming that's the proposed approach.)

Reply:
1. Once the fetcher service is complete and block importer is replaced with a chain manager
2. what outgoing messages do you mean? those that were previously going to BlockImporter will now go to the ChainManager, that's mentioned in the comments, unless you mean something else
3. of course we can try :) what logic from this actor would you extract?
4.  I'm not sure I follow here: move the code out where? the API won't change much indeed, it's just gonna be a bit simpler and message propagating will follow another path, but the function of this actor will stay largely the same (unless we decide to change that)

Reply:
>     2\. what outgoing messages do you mean? those that were previously going to BlockImporter will now go to the ChainManager, that's mentioned in the comments, unless you mean something else

What about the other `ProgressProtocol` messages and messages to the fetcher?


>     3\. of course we can try :) what logic from this actor would you extract?

I propose a straight forward transformation into a case class with values for every bit of state and transition methods for the different messages that cause state changes. A bit like `BlockFetcherState`.



>     4\. I'm not sure I follow here: move the code out where? the API won't change much indeed, it's just gonna be a bit simpler and message propagating will follow another path, but the function of this actor will stay largely the same (unless we decide to change that)

I mean to replace the current `RegularSync` actor gradually, instead of all at once.

Reply:
> What about the other ProgressProtocol messages and messages to the fetcher?

ah incoming messages you meant, I don't see a point in ProgressProtocol messages. do you? it's only used for logging, why not to log in place?

And the only message that was going to the fetcher was used to keep the fetcher state's blocks queue aware of the mined and checkpointed  blocks. This is no longer our intention with the new design.

> I propose a straight forward transformation into a case class with values for every bit of state and transition methods for the different messages that cause state changes. A bit like BlockFetcherState.

I still don't see what you're trying to achieve, BlockFetcherState is introduced for handling a state and a validation logic, while ExecutionSync/RegularSync is just an actor passing the messages around with no logic being executed inside an actor. 

But feel free to create your version of it, then we can discuss it.

> I meant to replace the current RegularSync actor gradually, instead of all at once.

We don't replace it atm, the decision was just to have this class in our codebase as a hint of how the things will be glued later, the transition of this part was not discussed yet.


I don't mind not merging this at all, if the purpose of it (or a gain) is not clear